system-metrics/basic_sender.go

```diff
@@ -0,0 +1,182 @@
+package systemmetrics
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/evergreen-ci/timber"
+	"github.com/evergreen-ci/timber/internal"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+// CompressionType describes how the system metrics data is compressed.
+type CompressionType int32
+
+// Valid CompressionType values.
+const (
+	CompressionTypeNone  CompressionType = 0
+	CompressionTypeTARGZ CompressionType = 1
+	CompressionTypeZIP   CompressionType = 2
+	CompressionTypeGZ    CompressionType = 3
+	CompressionTypeXZ    CompressionType = 4
+)
+
+func (f CompressionType) validate() error {
+	switch f {
+	case CompressionTypeNone, CompressionTypeTARGZ, CompressionTypeZIP, CompressionTypeGZ, CompressionTypeXZ:
+		return nil
+	default:
+		return errors.New("invalid compression type specified")
+	}
+}
+
+// SchemaType describes how the time series data is stored.
+type SchemaType int32
+
+// Valid SchemaType values.
+const (
+	SchemaTypeRawEvents             SchemaType = 0
+	SchemaTypeCollapsedEvents       SchemaType = 1
+	SchemaTypeIntervalSummarization SchemaType = 2
+	SchemaTypeHistogram             SchemaType = 3
+)
+
+func (f SchemaType) validate() error {
+	switch f {
+	case SchemaTypeRawEvents, SchemaTypeCollapsedEvents, SchemaTypeIntervalSummarization, SchemaTypeHistogram:
+		return nil
+	default:
+		return errors.New("invalid schema type specified")
+	}
+}
+
+type systemMetricsClient struct {
+	cancel context.CancelFunc
+	client internal.CedarSystemMetricsClient
+}
+
+type ConnectionOptions struct {
+	APIKey   string
+	Username string
+	Client   http.Client
+}
+
+func NewSystemMetricsClient(ctx context.Context, clientConn *grpc.ClientConn, opts ConnectionOptions) (*systemMetricsClient, error) {
```

Comment:
Split this into two functions `NewSystemMetricsClient` and `NewSystemMetricsClientWithExistingClient`